loupe ... does not provide an export named 'inspect'

### Describe the bug

Experiencing this on an update to 1.1.3:

```
Error: Vitest failed to load "vitest/utils" after 10 retries.
 ❯ WebSocket.<anonymous> ../../../../../../__vitest_browser__/index-CO7GSZ7y.js:964:36

Caused by: SyntaxError: The requested module '/@fs/path/to/my/package/node_modules/.pnpm/loupe@2.3.7/node_modules/loupe/loupe.js?v=12a7f0f8' does not provide an export named 'inspect'
```

When I downgrade both `vitest` and `@vitest/browser` packages, it still somehow finds 1.1.3 and has the same issue.

I encountered similar [issues/4384](https://github.com/vitest-dev/vitest/issues/4384). I attempted to clear everything:
- New git clone
- Cleared pmpm cache

...however the issue remains. Is there something else I need to clear? Or is this a legitimate bug?

### Reproduction

Nothing exceptional in the test config:

```
test: {
  browser: {
    provider: 'playwright',
    name: 'chromium',
    enabled: true,
    headless: true,
  },
}
```



### System Info

```shell
System:
    OS: macOS 14.2.1
    CPU: (8) arm64 Apple M2
    Memory: 388.22 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.0.0 - /opt/homebrew/bin/node
    Yarn: 1.22.21 - /opt/homebrew/bin/yarn
    npm: 10.2.5 - ~/Documents/path/to/my/package/node_modules/.bin/npm
    pnpm: 8.14.0 - ~/Library/pnpm/pnpm
    bun: 1.0.3 - ~/.bun/bin/bun
  Browsers:
    Chrome: 120.0.6099.129
    Safari: 17.2.1
  npmPackages:
    @vitejs/plugin-basic-ssl: ^1.0.2 => 1.0.2 
    @vitest/browser: ^1.1.3 => 1.1.3 
    vite: ^5.0.11 => 5.0.11 
    vitest: ^1.1.3 => 1.1.3
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/browser/src/node/esmInjector.ts
+++ b/packages/browser/src/node/esmInjector.ts
@@ -38,8 +38,6 @@ export function injectVitestModule(code: string, id: string, parse: PluginContex
 
   const hoistIndex = 0
 
-  let hasInjected = false
-
   // this will transform import statements into dynamic ones, if there are imports
   // it will keep the import as is, if we don't need to mock anything
   // in browser environment it will wrap the module value with "vitest_wrap_module" function
@@ -76,7 +74,6 @@ export function injectVitestModule(code: string, id: string, parse: PluginContex
   }
 
   function defineExport(position: number, name: string, local = name) {
-    hasInjected = true
     s.appendLeft(
       position,
       `\nObject.defineProperty(${viInjectedKey}, "${name}", `
@@ -170,7 +167,6 @@ export function injectVitestModule(code: string, id: string, parse: PluginContex
         // named hoistable/class exports
         // export default function foo() {}
         // export default class A {}
-        hasInjected = true
         const { name } = node.declaration.id
         s.remove(node.start, node.start + 15 /* 'export default '.length */)
         s.append(
@@ -180,7 +176,6 @@ export function injectVitestModule(code: string, id: string, parse: PluginContex
       }
       else {
         // anonymous default exports
-        hasInjected = true
         s.update(
           node.start,
           node.start + 14 /* 'export default'.length */,
@@ -196,13 +191,10 @@ export function injectVitestModule(code: string, id: string, parse: PluginContex
       s.remove(node.start, node.end)
       const importId = defineImportAll(node.source.value as string)
       // hoist re-exports near the defined import so they are immediately exported
-      if (node.exported) {
+      if (node.exported)
         defineExport(hoistIndex, node.exported.name, `${importId}`)
-      }
-      else {
-        hasInjected = true
+      else
         s.appendLeft(hoistIndex, `${viExportAllHelper}(${viInjectedKey}, ${importId});\n`)
-      }
     }
   }
 
@@ -244,11 +236,10 @@ export function injectVitestModule(code: string, id: string, parse: PluginContex
     },
   })
 
-  if (hasInjected) {
-    // make sure "__vi_injected__" is declared as soon as possible
-    s.prepend(`const ${viInjectedKey} = { [Symbol.toStringTag]: "Module" };\n`)
-    s.append(`\nexport { ${viInjectedKey} }`)
-  }
+  // make sure "__vi_injected__" is declared as soon as possible
+  // prepend even if file doesn't export anything
+  s.prepend(`const ${viInjectedKey} = { [Symbol.toStringTag]: "Module" };\n`)
+  s.append(`\nexport { ${viInjectedKey} }`)
 
   return {
     ast,

--- a/packages/browser/src/node/index.ts
+++ b/packages/browser/src/node/index.ts
@@ -65,18 +65,27 @@ export default (project: WorkspaceProject, base = '/'): Plugin[] => {
               'vitest/browser',
               'vitest/runners',
               '@vitest/utils',
+
+              // loupe is manually transformed
+              'loupe',
             ],
             include: [
               'vitest > @vitest/utils > pretty-format',
               'vitest > @vitest/snapshot > pretty-format',
               'vitest > diff-sequences',
-              'vitest > loupe',
               'vitest > pretty-format',
               'vitest > pretty-format > ansi-styles',
               'vitest > pretty-format > ansi-regex',
               'vitest > chai',
             ],
           },
+        }
+      },
+      transform(code, id) {
+        if (id.includes('loupe/loupe.js')) {
+          const exportsList = ['custom', 'inspect', 'registerConstructor', 'registerStringTag']
+          const codeAppend = exportsList.map(i => `export const ${i} = globalThis.loupe.${i}`).join('\n')
+          return `${code}\n${codeAppend}\nexport default globalThis.loupe`
         }
       },
       async resolveId(id) {

--- a/packages/browser/src/node/index.ts
+++ b/packages/browser/src/node/index.ts
@@ -1,7 +1,6 @@
 import { fileURLToPath } from 'node:url'
 
 import { resolve } from 'node:path'
-import { builtinModules } from 'node:module'
 import sirv from 'sirv'
 import type { Plugin } from 'vite'
 import type { WorkspaceProject } from 'vitest/node'
@@ -61,7 +60,6 @@ export default (project: WorkspaceProject, base = '/'): Plugin[] => {
               'vitest/runners',
             ],
             exclude: [
-              ...builtinModules,
               'vitest',
               'vitest/utils',
               'vitest/browser',

--- a/packages/utils/src/display.ts
+++ b/packages/utils/src/display.ts
@@ -1,6 +1,6 @@
 // since this is already part of Vitest via Chai, we can just reuse it without increasing the size of bundle
 // @ts-expect-error doesn't have types
-import { inspect as loupe } from 'loupe'
+import * as loupe from 'loupe'
 
 interface LoupeOptions {
   showHidden?: boolean | undefined
@@ -101,7 +101,7 @@ export function format(...args: unknown[]) {
 export function inspect(obj: unknown, options: LoupeOptions = {}): string {
   if (options.truncate === 0)
     options.truncate = Number.POSITIVE_INFINITY
-  return loupe(obj, options)
+  return loupe.inspect(obj, options)
 }
 
 export function objDisplay(obj: unknown, options: LoupeOptions = {}): string {

--- a/test/browser/package.json
+++ b/test/browser/package.json
@@ -16,6 +16,7 @@
     "@vitest/cjs-lib": "link:./cjs-lib",
     "execa": "^7.1.1",
     "safaridriver": "^0.0.4",
+    "url": "^0.11.3",
     "vitest": "workspace:*"
   }
 }

--- a/test/browser/package.json
+++ b/test/browser/package.json
@@ -5,7 +5,7 @@
   "scripts": {
     "test": "pnpm run test:webdriverio && pnpm run test:playwright",
     "test:webdriverio": "PROVIDER=webdriverio node --test --test-concurrency=1 specs/",
-    "test:playwright": "PROVIDER=playwright node --test --test-concurrency=1 specs/runner.test.mjs",
+    "test:playwright": "PROVIDER=playwright node --test --test-concurrency=1 specs/",
     "test:safaridriver": "PROVIDER=webdriverio BROWSER=safari node --test --test-concurrency=1 specs/",
     "test-fixtures": "vitest",
     "coverage": "vitest --coverage.enabled --coverage.provider=istanbul --browser.headless=yes"

--- a/test/browser/package.json
+++ b/test/browser/package.json
@@ -5,7 +5,7 @@
   "scripts": {
     "test": "pnpm run test:webdriverio && pnpm run test:playwright",
     "test:webdriverio": "PROVIDER=webdriverio node --test --test-concurrency=1 specs/",
-    "test:playwright": "PROVIDER=playwright node --test --test-concurrency=1 specs/",
+    "test:playwright": "PROVIDER=playwright node --test --test-concurrency=1 specs/runner.test.mjs",
     "test:safaridriver": "PROVIDER=webdriverio BROWSER=safari node --test --test-concurrency=1 specs/",
     "test-fixtures": "vitest",
     "coverage": "vitest --coverage.enabled --coverage.provider=istanbul --browser.headless=yes"

--- a/test/browser/specs/run-vitest.mjs
+++ b/test/browser/specs/run-vitest.mjs
@@ -20,7 +20,7 @@ export default async function runVitest(moreArgs = []) {
   const browserResult = await readFile('./browser.json', 'utf-8')
   const browserResultJson = JSON.parse(browserResult)
 
-  const getPassed = results => results.filter(result => result.status === 'passed')
+  const getPassed = results => results.filter(result => result.status === 'passed' && !result.mesage)
   const getFailed = results => results.filter(result => result.status === 'failed')
 
   const passedTests = getPassed(browserResultJson.testResults)

--- a/test/browser/specs/runner.test.mjs
+++ b/test/browser/specs/runner.test.mjs
@@ -11,9 +11,9 @@ const {
 } = await runVitest()
 
 await test('tests are actually running', async () => {
-  assert.ok(browserResultJson.testResults.length === 12, 'Not all the tests have been run')
-  assert.ok(passedTests.length === 10, 'Some tests failed')
-  assert.ok(failedTests.length === 2, 'Some tests have passed but should fail')
+  assert.equal(browserResultJson.testResults.length, 13, 'Not all the tests have been run')
+  assert.equal(passedTests.length, 11, 'Some tests failed')
+  assert.equal(failedTests.length, 2, 'Some tests have passed but should fail')
 
   assert.doesNotMatch(stderr, /Unhandled Error/, 'doesn\'t have any unhandled errors')
 })

--- a/test/browser/specs/runner.test.mjs
+++ b/test/browser/specs/runner.test.mjs
@@ -11,10 +11,11 @@ const {
 } = await runVitest()
 
 await test('tests are actually running', async () => {
-  assert.equal(browserResultJson.testResults.length, 13, 'Not all the tests have been run')
-  assert.equal(passedTests.length, 11, 'Some tests failed')
+  assert.equal(browserResultJson.testResults.length, 14, 'Not all the tests have been run')
+  assert.equal(passedTests.length, 12, 'Some tests failed')
   assert.equal(failedTests.length, 2, 'Some tests have passed but should fail')
 
+  assert.doesNotMatch(stderr, /has been externalized for browser compatibility/, 'doesn\'t have any externalized modules')
   assert.doesNotMatch(stderr, /Unhandled Error/, 'doesn\'t have any unhandled errors')
 })
 

--- a/test/browser/test/polyfill-lib.test.ts
+++ b/test/browser/test/polyfill-lib.test.ts
@@ -1,0 +1,7 @@
+// eslint-disable-next-line unicorn/prefer-node-protocol
+import * as url from 'url'
+import { expect, test } from 'vitest'
+
+test('url is polifylled because it\'s installed in dependencies', () => {
+  expect(url.format).toBeDefined()
+})

--- a/test/browser/test/utils.test.ts
+++ b/test/browser/test/utils.test.ts
@@ -1,0 +1,6 @@
+import { inspect } from 'vitest/utils'
+import { expect, it } from 'vitest'
+
+it('utils package correctly uses loupe', async () => {
+  expect(inspect({ test: 1 })).toBe('{ test: 1 }')
+})

--- a/test/core/test/injector-esm.test.ts
+++ b/test/core/test/injector-esm.test.ts
@@ -15,8 +15,10 @@ test('default import', async () => {
   expect(
     injectSimpleCode('import foo from \'vue\';console.log(foo.bar)'),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
-    console.log(__vi_esm_0__.default.bar)"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    console.log(__vi_esm_0__.default.bar)
+    export { __vi_inject__ }"
   `)
 })
 
@@ -26,8 +28,10 @@ test('named import', async () => {
       'import { ref } from \'vue\';function foo() { return ref(0) }',
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
-    function foo() { return __vi_esm_0__.ref(0) }"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    function foo() { return __vi_esm_0__.ref(0) }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -37,8 +41,10 @@ test('namespace import', async () => {
       'import * as vue from \'vue\';function foo() { return vue.ref(0) }',
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
-    function foo() { return __vi_esm_0__.ref(0) }"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    function foo() { return __vi_esm_0__.ref(0) }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -173,8 +179,10 @@ test('hoist import to top', async () => {
       'path.resolve(\'server.js\');import path from \'node:path\';',
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'node:path'
-    __vi_esm_0__.default.resolve('server.js');"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'node:path'
+    __vi_esm_0__.default.resolve('server.js');
+    export { __vi_inject__ }"
   `)
 })
 
@@ -200,8 +208,10 @@ test('do not rewrite method definition', async () => {
     'import { fn } from \'vue\';class A { fn() { fn() } }',
   )
   expect(result).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
-    class A { fn() { __vi_esm_0__.fn() } }"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    class A { fn() { __vi_esm_0__.fn() } }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -210,8 +220,10 @@ test('do not rewrite when variable is in scope', async () => {
     'import { fn } from \'vue\';function A(){ const fn = () => {}; return { fn }; }',
   )
   expect(result).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
-    function A(){ const fn = () => {}; return { fn }; }"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    function A(){ const fn = () => {}; return { fn }; }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -221,8 +233,10 @@ test('do not rewrite when variable is in scope with object destructuring', async
     'import { fn } from \'vue\';function A(){ let {fn, test} = {fn: \'foo\', test: \'bar\'}; return { fn }; }',
   )
   expect(result).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
-    function A(){ let {fn, test} = {fn: 'foo', test: 'bar'}; return { fn }; }"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    function A(){ let {fn, test} = {fn: 'foo', test: 'bar'}; return { fn }; }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -232,8 +246,10 @@ test('do not rewrite when variable is in scope with array destructuring', async 
     'import { fn } from \'vue\';function A(){ let [fn, test] = [\'foo\', \'bar\']; return { fn }; }',
   )
   expect(result).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
-    function A(){ let [fn, test] = ['foo', 'bar']; return { fn }; }"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    function A(){ let [fn, test] = ['foo', 'bar']; return { fn }; }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -244,8 +260,10 @@ test('rewrite variable in string interpolation in function nested arguments', as
     'import { fn } from \'vue\';function A({foo = `test${fn}`} = {}){ return {}; }',
   )
   expect(result).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
-    function A({foo = \`test\${__vi_esm_0__.fn}\`} = {}){ return {}; }"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    function A({foo = \`test\${__vi_esm_0__.fn}\`} = {}){ return {}; }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -255,8 +273,10 @@ test('rewrite variables in default value of destructuring params', async () => {
     'import { fn } from \'vue\';function A({foo = fn}){ return {}; }',
   )
   expect(result).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
-    function A({foo = __vi_esm_0__.fn}){ return {}; }"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    function A({foo = __vi_esm_0__.fn}){ return {}; }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -265,8 +285,10 @@ test('do not rewrite when function declaration is in scope', async () => {
     'import { fn } from \'vue\';function A(){ function fn() {}; return { fn }; }',
   )
   expect(result).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
-    function A(){ function fn() {}; return { fn }; }"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    function A(){ function fn() {}; return { fn }; }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -275,8 +297,10 @@ test('do not rewrite catch clause', async () => {
     'import {error} from \'./dependency\';try {} catch(error) {}',
   )
   expect(result).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from './dependency'
-    try {} catch(error) {}"
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from './dependency'
+    try {} catch(error) {}
+    export { __vi_inject__ }"
   `)
 })
 
@@ -287,9 +311,11 @@ test('should declare variable for imported super class', async () => {
       'import { Foo } from \'./dependency\';' + 'class A extends Foo {}',
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from './dependency'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from './dependency'
     const Foo = __vi_esm_0__.Foo;
-    class A extends Foo {}"
+    class A extends Foo {}
+    export { __vi_inject__ }"
   `)
 
   // exported classes: should prepend the declaration at root level, before the
@@ -375,7 +401,8 @@ test('overwrite bindings', async () => {
         + 'function g() { const f = () => { const inject = true }; console.log(inject) }\n',
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
     const a = { inject: __vi_esm_0__.inject }
     const b = { test: __vi_esm_0__.inject }
     function c() { const { test: inject } = { test: true }; console.log(inject) }
@@ -383,14 +410,19 @@ test('overwrite bindings', async () => {
     function f() {  console.log(__vi_esm_0__.inject) }
     function e() { const { inject } = { inject: true } }
     function g() { const f = () => { const inject = true }; console.log(__vi_esm_0__.inject) }
-    "
+
+    export { __vi_inject__ }"
   `)
 })
 
 test('Empty array pattern', async () => {
   expect(
     injectSimpleCode('const [, LHS, RHS] = inMatch;'),
-  ).toMatchInlineSnapshot('"const [, LHS, RHS] = inMatch;"')
+  ).toMatchInlineSnapshot(`
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    const [, LHS, RHS] = inMatch;
+    export { __vi_inject__ }"
+  `)
 })
 
 test('function argument destructure', async () => {
@@ -404,13 +436,15 @@ function c({ _ = bar() + foo() }) {}
 `,
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'foo'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'foo'
 
 
     const a = ({ _ = __vi_esm_0__.foo() }) => {}
     function b({ _ = __vi_esm_0__.bar() }) {}
     function c({ _ = __vi_esm_0__.bar() + __vi_esm_0__.foo() }) {}
-    "
+
+    export { __vi_inject__ }"
   `)
 })
 
@@ -426,14 +460,16 @@ const a = () => {
 `,
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'foo'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'foo'
 
 
     const a = () => {
       const { type: n = 'bar' } = {}
       console.log(n)
     }
-    "
+
+    export { __vi_inject__ }"
   `)
 
   // #9585
@@ -449,7 +485,8 @@ const foo = {}
 `,
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'foo'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'foo'
 
 
     const foo = {}
@@ -457,7 +494,8 @@ const foo = {}
     {
       const { [__vi_esm_0__.n]: m } = foo
     }
-    "
+
+    export { __vi_inject__ }"
   `)
 })
 
@@ -492,7 +530,8 @@ objRest()
 `,
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
 
 
 
@@ -518,7 +557,8 @@ objRest()
     __vi_esm_0__.set()
     __vi_esm_0__.rest()
     __vi_esm_0__.objRest()
-    "
+
+    export { __vi_inject__ }"
   `)
 })
 
@@ -542,7 +582,8 @@ const obj = {
 `,
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'foo'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'foo'
 
 
 
@@ -557,7 +598,8 @@ const obj = {
       [bar]: () => {},
       bar(foo) {}
     }
-    "
+
+    export { __vi_inject__ }"
   `)
 })
 
@@ -574,7 +616,8 @@ class A {
 `,
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
 
 
 
@@ -584,7 +627,8 @@ class A {
       remove = 1
       add = null
     }
-    "
+
+    export { __vi_inject__ }"
   `)
 })
 
@@ -606,7 +650,8 @@ class A {
 `,
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'foo'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'foo'
 
 
 
@@ -619,7 +664,8 @@ class A {
       #foo() {}
       bar(foo) {}
     }
-    "
+
+    export { __vi_inject__ }"
   `)
 })
 
@@ -652,7 +698,8 @@ bbb()
 `,
     ),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'vue'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'vue'
 
 
 
@@ -676,7 +723,8 @@ bbb()
 
     __vi_esm_0__.aaa()
     __vi_esm_0__.bbb()
-    "
+
+    export { __vi_inject__ }"
   `)
 })
 
@@ -697,14 +745,16 @@ test('jsx', async () => {
   const result = await transformWithEsbuild(code, id)
   expect(injectSimpleCode(result.code))
     .toMatchInlineSnapshot(`
-      "import { __vi_inject__ as __vi_esm_0__ } from 'react'
+      "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+      import { __vi_inject__ as __vi_esm_0__ } from 'react'
       import { __vi_inject__ as __vi_esm_1__ } from 'foo'
 
 
       function Bar({ Slot: Slot2 = /* @__PURE__ */ __vi_esm_0__.default.createElement(__vi_esm_1__.Foo, null) }) {
         return /* @__PURE__ */ __vi_esm_0__.default.createElement(__vi_esm_0__.default.Fragment, null, /* @__PURE__ */ __vi_esm_0__.default.createElement(Slot2, null));
       }
-      "
+
+      export { __vi_inject__ }"
     `)
 })
 
@@ -777,7 +827,8 @@ for (const test in tests) {
   console.log(test)
 }`),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from './test.js'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from './test.js'
 
 
     for (const test of tests) {
@@ -788,7 +839,8 @@ for (const test in tests) {
     }
     for (const test in tests) {
       console.log(test)
-    }"
+    }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -892,7 +944,8 @@ function test() {
   return [foo, bar]
 }`),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'foobar'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'foobar'
 
 
     function test() {
@@ -900,7 +953,8 @@ function test() {
         var foo = () => { var why = 'would' }, bar = 'someone'
       }
       return [foo, bar]
-    }"
+    }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -919,7 +973,8 @@ function test() {
   return bar;
 }`),
   ).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from 'foobar'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from 'foobar'
 
 
     function test() {
@@ -930,7 +985,8 @@ function test() {
       }
       try {} catch (baz){ baz };
       return __vi_esm_0__.bar;
-    }"
+    }
+    export { __vi_inject__ }"
   `)
 })
 
@@ -947,7 +1003,8 @@ test('avoid binding ClassExpression', () => {
  `,
   )
   expect(result).toMatchInlineSnapshot(`
-    "import { __vi_inject__ as __vi_esm_0__ } from './foo'
+    "const __vi_inject__ = { [Symbol.toStringTag]: "Module" };
+    import { __vi_inject__ as __vi_esm_0__ } from './foo'
 
      
      console.log(__vi_esm_0__.default, __vi_esm_0__.Bar);
@@ -956,6 +1013,7 @@ test('avoid binding ClassExpression', () => {
        bar: class Bar {}
      }
      const Baz = class extends __vi_esm_0__.default {}
-     "
+     
+    export { __vi_inject__ }"
   `)
 })

